conda/cli/main_env_create.py

```diff
@@ -141,70 +154,97 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
     prefix_data.validate_path()
     prefix_data.validate_name()
 
-    # TODO, add capability
-    # common.ensure_override_channels_requires_channel(args)
-    # channel_urls = args.channel or ()
-
-    result = {"conda": None, "pip": None}
+    # Convert all environment configuration to environment variables for fresh_context
+    # NOTE: Only channels and environment variables are currently supported in
+    # environment.yaml files.
+    env_vars = {}
+    # Add environment variables from YAML (convert all values to strings)
+    if env.variables:
+        env_vars.update({k: str(v) for k, v in env.variables.items()})
+
+    if env.config.channels:
+        env_vars["CONDA_CHANNELS"] = ",".join(env.config.channels)
```

Comment:
This needs to work for all the configuration options in the `EnvironmentConfig` object. And should be extensible so that developers can add more configuration options in the future.

For example, the environment spec api supports specs that can set any of the configuration  defined in `EnvironmentConfig`. If a spec decides it needs that config to exist, then conda should respect that.